src/cart/crt_hg_proc.c

```diff
@@ -412,9 +435,23 @@ crt_hg_unpack_header(hg_handle_t handle, struct crt_rpc_priv *rpc_priv,
 		D_ERROR("crt_proc_common_hdr failed rc: %d.\n", rc);
 		D_GOTO(out, rc);
 	}
-	/* Clients never decode requests. */
+
+	/* Sync the HLC. Clients never decode requests. */
 	D_ASSERT(crt_is_service());
-	(void)crt_hlc_get_msg(rpc_priv->crp_req_hdr.cch_hlc);
+	rc = crt_hlc_get_msg(rpc_priv->crp_req_hdr.cch_hlc, NULL /* hlc_out */);
+	if (rc != 0) {
+		REPORT_HLC_SYNC_ERR("failed to sync HLC for request: opc=%x ts="
+				    DF_U64" from=%u\n",
+				    rpc_priv->crp_req_hdr.cch_opc,
+				    rpc_priv->crp_req_hdr.cch_hlc,
+				    rpc_priv->crp_req_hdr.cch_src_rank);
+		/* Fail all but SWIM requests. */
+		if (crt_opc_is_swim(rpc_priv->crp_req_hdr.cch_opc))
+			rc = 0;
+		else
+			D_GOTO(out, rc);
```

Comment:
Maybe the following construction will be better?
```
if (!crt_opc_is_swim(rpc_priv->crp_req_hdr.cch_opc))
	D_GOTO(out, rc);
rc = 0;
```

Reply:
OK. Changed.